autoflake.py

```diff
@@ -619,9 +619,14 @@ def fix_code(source, additional_imports=None, expand_star_imports=False,
 
 def fix_file(filename, args, standard_out):
     """Run fix_code() on a file."""
-    encoding = detect_encoding(filename)
-    with open_with_encoding(filename, encoding=encoding) as input_file:
+
+    if filename == '-':
```

Comment:
`fix_file()` is part of the public API, so we probably shouldn't parse stdin here. It would be better to do this in the main program so that we don't break usage for other programs that are using `autoflake.fix_file()`. It would be weird for them if a GUI program, for example, started trying to read form stdin.

Example:

https://github.com/myint/docformatter/blob/7a23f89a8bcfb117db2f388c6706d28e9bb30e9c/docformatter.py#L593

Reply:
I don't see how this could break other programs since a program can't read from STDIN without explicitly passing '-' as a filename, which is a well known UNIX convention for reading STDIN anyway.

Reply:
It is a well-known convention, but the convention is for the command line. Not the GUI.

For example:

- GUI program tries to open a file called `-`.
- It hangs forever waiting for a non-existant input from stdin.

Basically, the intent is for `_main()` to contain command-line specific code. And `fix_file()` to contain more general code that can be used by as a library function by other programs.